test/integration/crud/explain.test.ts

```diff
@@ -0,0 +1,250 @@
+    client = this.configuration.newClient({ monitorCommands: true });
+    db = client.db('queryPlannerExplainResult');
+    collection = db.collection('test');
+
+    await collection.insertOne({ a: 1 });
+  });
+
+  afterEach(async function () {
+    await collection.drop();
+    await client.close();
+  });
+
+  context('when explain is set to true', () => {
+    it('deleteOne returns queryPlanner explain result', async function () {
+      const explanation = await collection.deleteOne({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('deleteMany returns queryPlanner explain result', async function () {
+      const explanation = await collection.deleteMany({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('updateOne returns queryPlanner explain result', async function () {
+      const explanation = await collection.updateOne(
+        { a: 1 },
+        { $inc: { a: 2 } },
+        { explain: true }
+      );
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('updateMany returns queryPlanner explain result', async function () {
+      const explanation = await collection.updateMany(
+        { a: 1 },
+        { $inc: { a: 2 } },
+        { explain: true }
+      );
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('distinct returns queryPlanner explain result', async function () {
+      const explanation = await collection.distinct('a', {}, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('findOneAndDelete returns queryPlanner explain result', async function () {
+      const explanation = await collection.findOneAndDelete({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('allPlansExecution returns verbose queryPlanner explain result', async function () {
+      const explanation = await collection.deleteOne({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+      expect(explanation).nested.property('executionStats.allPlansExecution').to.exist;
+    });
+
+    it('findOne returns queryPlanner explain result', async function () {
+      const explanation = await collection.findOne({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('find returns queryPlanner explain result', async () => {
+      const [explanation] = await collection.find({ a: 1 }, { explain: true }).toArray();
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+  });
+
+  context('when explain is set to false', () => {
+    it('only queryPlanner property is used in explain result', async function () {
+      const explanation = await collection.deleteOne({ a: 1 }, { explain: false });
+      expect(explanation).property('queryPlanner').to.exist;
```

Comment:
Two comments on this test - first, the title of the test doesn't reflect the actual api used (`deleteOne`) - this would be fine if the assertion is intended to be generic across all APIs, but if that were the case, we would want to call at least a couple of different APIs and assert on the outcome of each. Second, it's odd to me that the assertion in the "when explain is set to true" and "when explain is set to false" is identical (as is the "when explain is set to queryPlanner" lower down). Is the behavior actually identical for all these values of explain? If not, then we should make sure that the assertions clearly differentiate the expected behavior.

Reply:
The reason we expect that the same behaviour for `explain=false` and `explain='queryPlanner'` is that we [interpret our boolean values](https://github.com/mongodb/node-mongodb-native/blob/dc068ac1f6e91794836e103f8a06c2580405ad76/src/explain.ts#L32-L40) of explain more like numeric verbosity with `false` corresponding to the lowest verbosity which is `queryPlanner` and `true` to the highest which is `allPlansExecution`. It's admittedly a strange choice, but it's a choice that got made 😅

Reply:
@W-A-James That's very good context - thank you - I think we can capture that in the test descriptions (or comments) to avoid any similar confusion in the future. This could be done, for example, via a wrapper `context('lowest verbosity')` and `context('highest verbosity')` for the other groupings